projects/plugins/automattic-for-agencies-client/automattic-for-agencies-client.php

```diff
@@ -117,5 +117,14 @@ function jetpack_starter_plugin_activation( $plugin ) {
 
 register_deactivation_hook( __FILE__, array( 'Automattic_For_Agencies_Client', 'plugin_deactivation' ) );
 
-// Main plugin class.
-new Automattic_For_Agencies_Client();
+/**
+ * Initialize the plugin's main class.
+ *
+ * @since $$next-version$$
+ *
+ * @return void
+ */
+function automattic_for_agencies_run_plugin() {
+	new Automattic_For_Agencies_Client();
```

Comment:
Phan's message does note that it might be being called for side effects of the constructor, which is very likely the case.

In this case it looks like we might do better to move what's in `__construct()` to a static `Automattic_For_Agencies_Client::init()` method instead, since nothing in there saves a reference to the new instance.

Reply:
Done in ea021c469aa62527103f6f2418c70b092f040e4c

Reply:
I note ea021c4 is a slight behavior change, in that all the `::init()` stuff is now done in the middle of loading plugins instead of waiting for the 'plugins_loaded' action. Does that matter?

Reply:
I'm happy to bring the `plugins_loaded` hook back, I think it lines up a bit more with how we do things in the Jetpack plugin. 7859ab333bcf1f711feed2ac08503cde334929c5